include/cppkafka/utils/poll_strategy_base.h

```diff
@@ -84,6 +84,29 @@ class CPPKAFKA_API PollStrategyBase : public PollInterface {
      */
     Consumer& get_consumer() final;
     
+    /**
+     * \brief Creates partitions queues associated with the supplied partitions.
+     *
+     * This method contains a default implementation. It adds all the new queues belonging
+     * to the provided partition list and calls reset_state().
+     * To be used with static consumers.
+     *
+     * \param partitions Assigned topic partitions.
+     */
+    virtual void assign(TopicPartitionList& partitions);
+    
+    /**
+     * \brief Removes partitions queues associated with the supplied partitions.
+     *
+     * This method contains a default implementation. It removes all the queues
+     * belonging to the provided partition list and calls reset_state().
+     * To be used with static consumers.
+     *
+     * \param partitions Revoked topic partitions. If the partition list is empty
+     *                   all partitions will be revoked.
```

Comment:
What's the reason for this behavior of removing all if the list is empty? That feels a bit counter intuitive. You might as well add a `revoke()` that takes nothing which makes the semantics a bit more clear from the user's perspective